### PR TITLE
R Syntax: split analysis and analysisForm

### PR DESCRIPTION
**Pull request by boutinb**
As described in the JASP Architecture for R Syntax document, we need to split the analysis and analysisForm objects. For this the AnalysisFormBase is added (AnalysisBase existed already), the Base objects contain all calls from Analysis to AnalysisForm and vice-versa.

The initColumnWithString is function is moved to the DataSet object so that the RDataFrame importer can use it.

The parseOptions function in the AnalysisForm is already added.

### CODE REVIEW

File: QMLComponents/analysisform.cpp [hunk 1]
```diff
@@ -272,16 +273,28 @@ void AnalysisForm::setHasVolatileNotes(bool hasVolatileNotes)
...
-		options = _analysis->boundValues();
-		return true;
+		bindTo(jsonOptions);
+		jsonOptions = _analysis->boundValues();
 	}
 
-	return false;
+	jsonResult["options"] = jsonOptions;
+	jsonResult["error"] = fq(getError());
+	return tq(jsonResult.toStyledString());
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.cpp:292]: Why not return the Json value then?

Comment (boutinb) [reply QMLComponents/analysisform.cpp:292]: In the jaspQMLR example, the QML form is just a QuickItem object, so I call the parseOptions with the QMetaObject::invokeMethod method:
https://github.com/jasp-stats/jaspQmlR/blob/a7a93a2f6fa103d47560c1ed541e9c252eed3671/src/qmlR.cpp#L208
In this way I don't need to load the AnalysisFormBase header in the jaspQMLR package.
I don't think I can pass a Json::Value argument in the InvokeMethod, so I thought just send the string, and reset it as Json object afterwards.

File: QMLComponents/analysisform.cpp [hunk 2]
```diff
@@ -272,16 +270,28 @@ void AnalysisForm::setHasVolatileNotes(bool hasVolatileNotes)
...
-bool AnalysisForm::parseOptions(Json::Value &options)
+
+QString AnalysisForm::parseOptions(QString options)
 {
-	if (_rSyntax->parseRSyntaxOptions(options))
+	Json::Reader jsonReader;
+	Json::Value	 jsonOptions;
+	Json::Value jsonResult(Json::objectValue);
+
+	QJsonDocument doc = QJsonDocument::fromJson(options.toUtf8());
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.cpp]: Why is this step here?
It seems that the whole QJsonDocument

File: QMLComponents/analysisform.cpp [hunk 3]
```diff
@@ -135,11 +133,10 @@ void AnalysisForm::runScriptRequestDone(const QString& result, const QString& co
 				// Some controls generate extra controls (rowComponents): these extra controls must be first destroyed, because they may disturb the binding of other controls
 				// For this, bind all controls to null and wait for the controls to be completely destroyed.
 				QTimer::singleShot(0, [=](){
-					bindTo(options);
+				bindTo(options);
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.cpp]:
```suggestion
					bindTo(options);
```

File: QMLComponents/analysisform.cpp [hunk 4]
```diff
@@ -135,11 +133,10 @@ void AnalysisForm::runScriptRequestDone(const QString& result, const QString& co
...
 					blockValueChangeSignal(false, false);
-					_analysis->boundValueChangedHandler();
+				_analysis->boundValueChangedHandler();
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.cpp]:
```suggestion
					_analysis->boundValueChangedHandler();
```

File: QMLComponents/analysisform.cpp [hunk 5]
```diff
@@ -567,7 +575,7 @@ void AnalysisForm::setTitle(QString title)
 		_analysis->setTitle(fq(title.simplified()));
 }
 
-void AnalysisForm::setOptionNameConversion(const QVariantList & conv)
+void AnalysisForm::setOptionNameConversion(const QVariantList &conv)
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.cpp]:
```suggestion
void AnalysisForm::setOptionNameConversion(const QVariantList & conv)
```

File: QMLComponents/analysisform.h [hunk 6]
```diff
@@ -145,6 +145,7 @@ public slots:
 	Q_INVOKABLE void		refreshAnalysis();
 	Q_INVOKABLE bool		initialized()			const	{ return _initialized; }
 	Q_INVOKABLE QString		generateWrapper()		const;
+    Q_INVOKABLE QString     parseOptions(QString options);
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.h]:
```suggestion
	Q_INVOKABLE QString		parseOptions(QString options);
```

File: QMLComponents/analysisform.h [hunk 7]
```diff
@@ -180,9 +181,8 @@ public slots:
 	void					sortControls(QList<JASPControl*>& controls);
 	QString					getSyntaxName(const QString& name)				const;
 	void					setHasVolatileNotes(bool hasVolatileNotes);
-	bool					parseOptions(Json::Value& options);
-	void					setActiveJASPControl(JASPControl* control, bool hasActiveFocus);
-	JASPControl*			getActiveJASPControl()	{ return _activeJASPControl; }
+    void					setActiveJASPControl(JASPControl* control, bool hasActiveFocus);
+    JASPControl*			getActiveJASPControl()	{ return _activeJASPControl; }
```
Comment (JorisGoosen) [inline QMLComponents/analysisform.h]:
```suggestion
	void					setActiveJASPControl(JASPControl* control, bool hasActiveFocus);
	JASPControl*			getActiveJASPControl()	{ return _activeJASPControl; }
```

**Review by JorisGoosen — APPROVED** [review]

**Pull request merged**